sklearn/utils/estimator_checks.py

```diff
@@ -533,6 +547,11 @@ def parametrize_with_checks(estimators):
 
         .. versionadded:: 0.24
 
+    legacy : bool (default=True)
+        Whether to include legacy checks.
```

Comment:
I'm wondering if you should add a little note to mention that the legacy checks exists during the transition that we create the categorisation.